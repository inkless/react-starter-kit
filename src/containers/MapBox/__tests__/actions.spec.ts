import configureStore from 'redux-mock-store';

import {
  fetchOriginGeocodingAction,
  setViewport,
  setDeliverSimulationStarted,
  setDeliverGeoFeatures,
} from '../actions';
import { fixtureData1 } from './__fixture__/geojson';

const mockStore = configureStore();
const store = mockStore();

describe('MapBox Actions', () => {
  beforeEach(() => {
    store.clearActions();
  });

  it('should dispatch fetchOriginGeocodingAction', () => {
    store.dispatch(
      fetchOriginGeocodingAction.success({
        address: 'Home',
        coordinates: [-137, 37],
      })
    );
    expect(store.getActions()).toMatchSnapshot();
  });

  it('should dispatch setViewport', () => {
    store.dispatch(
      setViewport({
        longitude: -137,
        latitude: 37,
        zoom: 11,
      })
    );
    expect(store.getActions()).toMatchSnapshot();
  });

  it('should dispatch setDeliverSimulationStarted', () => {
    store.dispatch(setDeliverSimulationStarted(true));
    expect(store.getActions()).toMatchSnapshot();
  });

  it('should dispatch setDeliverGeoFeatures', () => {
    store.dispatch(setDeliverGeoFeatures([fixtureData1.route]));
    expect(store.getActions()).toMatchSnapshot();
  });
});
