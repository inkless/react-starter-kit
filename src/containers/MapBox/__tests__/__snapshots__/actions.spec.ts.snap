// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MapBox Actions should dispatch fetchOriginGeocodingAction 1`] = `
Array [
  Object {
    "payload": Object {
      "address": "Home",
      "coordinates": Array [
        -137,
        37,
      ],
    },
    "type": "@@MapBox/FETCH_ORIGIN_GEOCODING_SUCCESS",
  },
]
`;

exports[`MapBox Actions should dispatch setDeliverGeoFeatures 1`] = `
Array [
  Object {
    "meta": undefined,
    "payload": Array [
      Object {
        "geometry": Object {
          "coordinates": Array [
            Array [
              -118.46742,
              34.00534,
            ],
            Array [
              -118.47105,
              34.00321,
            ],
            Array [
              -118.47115,
              34.00309,
            ],
            Array [
              -118.47325,
              34.00463,
            ],
            Array [
              -118.47629,
              34.00289,
            ],
            Array [
              -118.48058,
              33.99993,
            ],
            Array [
              -118.48113,
              33.99954,
            ],
            Array [
              -118.48088,
              33.9993,
            ],
            Array [
              -118.48088,
              33.99928,
            ],
          ],
          "type": "LineString",
        },
        "properties": Object {},
        "type": "Feature",
      },
    ],
    "type": "@@MapBox/SET_DELIVER_GEO_FEATURES",
  },
]
`;

exports[`MapBox Actions should dispatch setDeliverSimulationStarted 1`] = `
Array [
  Object {
    "meta": undefined,
    "payload": true,
    "type": "@@MapBox/SET_DELIVER_SIMULATION_STARTED",
  },
]
`;

exports[`MapBox Actions should dispatch setViewport 1`] = `
Array [
  Object {
    "meta": undefined,
    "payload": Object {
      "latitude": 37,
      "longitude": -137,
      "zoom": 11,
    },
    "type": "@@MapBox/SET_VIEWPORT",
  },
]
`;
